modules/@apostrophecms/schema/index.js

```diff
@@ -547,18 +551,30 @@ module.exports = {
 
       async isVisible(req, schema, object, name) {
         const conditionalFields = {};
+        const errors = {};
+
         while (true) {
           let change = false;
           for (const field of schema) {
             if (field.if) {
-              const result = await evaluate(field.if, field.name, field.moduleName);
-              const previous = conditionalFields[field.name];
-              if (previous !== result) {
-                change = true;
+              try {
+                const result = await evaluate(field.if, field.name, field.moduleName);
+                const previous = conditionalFields[field.name];
+                if (previous !== result) {
+                  change = true;
+                }
+                conditionalFields[field.name] = result;
+              } catch (error) {
+                errors[field.name] = error;
               }
-              conditionalFields[field.name] = result;
             }
           }
+
+          // send the error related to the given field via the `name` param
+          if (errors[name]) {
+            throw errors[name];
+          }
```

Comment:
otherwise, we get the first error found in the fields of the schema, not the error related to the field name given in parameters, and that's very confusing.

now we get the correct error for the field that's required:

![image](https://user-images.githubusercontent.com/8301962/224100093-abdf2607-c40a-4e45-9938-4396498afd0e.png)